cmd/email-provider/server/operations.go

```diff
@@ -16,7 +20,69 @@ func NewServer() *Server {
 }
 
 func (server *Server) SendMessage(ctx context.Context, message *email_provider.SendEmailRequest) (*email_provider.SendEmailResponse, error) {
-	// TODO: send an email to someone here
+	m := mail.NewV3Mail()
+	address := os.Getenv("FROM_EMAIL_ADDRESS")
+	name := os.Getenv("FROM_NAME")
+	e := mail.NewEmail(name, address)
+	m.SetFrom(e)
+
+	p := mail.NewPersonalization()
+	tos := []*mail.Email{
+		mail.NewEmail(os.Getenv("TO_NAME"), os.Getenv("TO_EMAIL_ADDRESS")),
+	}
+	p.AddTos(tos...)
+	p.Subject = os.Getenv("SUBJECT")
+	m.AddPersonalizations(p)
+
+	htmlContent := fmt.Sprintf(
+		`
+		 <strong> First name: </strong> %s <br> 
+		 <strong> Last name: </strong> %s <br>
+		 <strong> Email: </strong> %s <br> 
+		 <strong> Phone: </strong> %s <br><br> 
+		 <strong> Request details: </strong> %s <br> 
+		 <strong> Request service: </strong> %s
+		`,
+		message.Sender.FirstName, message.Sender.LastName, message.Sender.Email, message.Sender.PhoneNumber, message.RequestDetails, message.RequestService)
+	c := mail.NewContent("text/html", htmlContent)
+	m.AddContent(c)
+
+	mailSettings := mail.NewMailSettings()
+	bypassBounceManagement := mail.NewSetting(true)
+	mailSettings.SetBypassBounceManagement(bypassBounceManagement)
+	bypassSpamManagement := mail.NewSetting(true)
+	mailSettings.SetBypassSpamManagement(bypassSpamManagement)
+	bypassUnsubscribeManagement := mail.NewSetting(true)
+	mailSettings.SetBypassUnsubscribeManagement(bypassUnsubscribeManagement)
+
+	footerSetting := mail.NewFooterSetting()
+	footerSetting.SetText("footer")
+	footerSetting.SetEnable(true)
+	footerSetting.SetHTML("<html><body><br><br>Club CEDILLE, ETS</body></html>")
```

Comment:
Pas obligé mais ça serait cool de voir si ça serait possible de faire quelque chose avec [text/template](https://pkg.go.dev/text/template) ici

Reply:
Ouais, faudrait voir. Pour la première itération, on pourrait garder ça. Aussi, on peut aussi faire un template via le site de SendGrid (https://sendgrid.com/free-template-builder/)

Reply:
ça peut tjrs s'améliorer dans le futur, tant qu'on a un envoi de courriel, c'est ça qui est important pour commencer